### DISCUSSION TITLE
#Resolved - magick -script and mogrify

### DISCUSSION THREAD

**Discussion by blue13821** [ImageMagick/ImageMagick#5087] [Help] [answered]
Hi,
I'im just discovering IM7. I was used to deal with IM6 but i need to use script and IM7 -script option appears to be a good option for me. At least i'm just a newbee for coding and i found very difficult to find information i can understand. I hope you can help me.
I'm using IM7 on my Ubuntu 21.10 distro.
I used to use mogrify to modify images.
On IM6, this code works fine :
```
convert -density 300 -size 200x200 xc:white fond.png
mogrify -fill blue -draw "rectangle 50,50,60,60" fond.png
```
When passing to IM7, this works fine too :
```
magick -density "300" -size 100x100 xc:cyan fond1.png
magick fond1.png -fill orange -draw "rectangle 20,20,40,40" fond2.png
magick -density 300 -size 100x100 xc:cyan fond3.png
magick fond3.png -fill blue -draw "rectangle 20,20,40,40" fond3.png
```
Then i intend to use -script option and nothing works fine. This do not work :
`echo "-density 300 -size 100x100 xc:cyan fond4.png" | magick -script -`
but this works :
```
echo "-density 300 -size 100x100 xc:cyan -write fond4.png" | magick -script -
echo "fond4.png -fill orange -draw \"rectangle 20,20,40,40\" -draw \"text 60,60 \"Text\"\" -write fond5.png" | magick -script -
echo "fond4.png -fill orange -draw \"rectangle 20,20,40,40\" -draw \"text 60,60 \"Text\"\" -write fond4.png" | magick -script -
```
So i supposed i have now to use -write option. OK. This works too :
`magick mogrify -fill orange -draw "rectangle 20,20,40,40" -draw "text 60,60 \"Text\"" fond7.png`
but this do not works :
`echo "mogrify -fill orange -draw "rectangle 20,20,40,40" -draw \"text 60,60 \"Text\"\" fond7.png" | magick -script - `
Does this means that it is not possible to use mogrify inside script ? Is applying magick to the same file in and out as i did for fond3.png and fond4.png the good way to replace IM6 mogrify with magick ?
Really thanks for helping.

**Comment by fmw42** [comment]
I do not use Magick Script, so cannot comment. But here is the link I have to what docs Anthony made:  https://legacy.imagemagick.org/Usage/bugs/IMv7_Scripting.txt

> convert -density 300 -size 200x200 xc:white fond.png
> mogrify -fill blue -draw "rectangle 50,50,60,60" fond.png

But why not just:

`magick -density 300 -size 200x200 xc:white -fill blue -draw "rectangle 50,50,60,60" fond.png`

**Reply by blue13821** [reply]
Thanks a lot. Yes the doc Anthony made helped me to begin but i didn't found all the replies i was looking for.
The example i gave can be improved the way you suggest, in command line. That is what i'm going to do but i had a more general answer about how to modify an existing image. It doesn't work in script, file created is empty. I must separates the two instructions :
```
#!/usr/bin/env magick-script
#
-density 300 -size 200x200 xc:green -write fond3.png
-fill blue -draw "rectangle 50,50,60,60" -write fond3.png
```
`magick -script img.script `

Snigbo gave me the reply : it's not possible to use mogrify in magick script. So i have to modify my way of thinking to manage with that and to go further the way you said.

**Comment by snibgo** [comment, marked as answer]
"-script" doesn't work with "magick mogrify", either by starting the script with "mogrify", or by running "magick mogrify -script ..." .

**Reply by blue13821** [reply]
Thanks. That's clear. That allows me to stop trying this way and to find another way.
